llmops/common/prompt_pipeline.py

```diff
@@ -230,47 +240,47 @@ def prepare_and_execute(
                             logger.info(df_result.head(10))
                         else:
                             raise Exception("Sorry, job failured..")
-        else:
-            timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
-            run = Run(
-                flow=flow,
-                data=data_id,
-                # runtime=runtime,
-                # un-comment the resources line and
-                # comment the argument runtime to
-                # enable automatic runtime.
-                # Reference: COMPUTE_RUNTIME
-                resources={"instance_type": "Standard_E4ds_v4"},
-                name=f"{experiment_name}_{timestamp}_{data_ref}",
-                display_name=f"{experiment_name}_{timestamp}_{data_ref}",
-                environment_variables={
-                    "key1": "value1"
-                    },
-                column_mapping=exp_config_node,
-                tags={
-                    "build_id": build_id
-                    },
-            )
-            run._experiment_name = experiment_name
-            pipeline_job = pf.runs.create_or_update(run, stream=True)
-            run_ids.append(pipeline_job.name)
-            time.sleep(15)
-            df_result = None
-    
-            if (pipeline_job.status == "Completed" or
-                    pipeline_job.status == "Finished"):
-
-                logger.info("job completed")
-                df_result = pf.get_details(pipeline_job)
-                if save_output:
-                    dataframes.append(df_result)
-                if save_metric:
-                    metric_variant = pf.get_metrics(pipeline_job)
-                    metric_variant["dataset"] = data_id
-                    metrics.append(metric_variant)
-                logger.info(df_result.head(10))
             else:
-                raise Exception("Sorry, exiting job with failure..")
+                timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+                run = Run(
+                    flow=flow,
+                    data=data_id,
+                    runtime=runtime,
+                    # un-comment the resources line and
+                    # comment the argument runtime to
+                    # enable automatic runtime.
+                    # Reference: COMPUTE_RUNTIME
+                    # resources={"instance_type": "Standard_E4ds_v4"},
+                    name=f"{experiment_name}_{rule}_{timestamp}_{data_ref}",
+                    display_name=f"{experiment_name}_{rule}_{timestamp}_{data_ref}",
+                    environment_variables={
+                        "key1": "value1"
+                        },
```

Comment:
is this property required? If there aren't any ENV Vars can this be left as an empty JSON object?

Reply:
done